### PR TITLE
DOV-6868: Port config-rewrite changes - Part VI

### PR DESCRIPTION
**Pull request by fleischie**
Another batch of changes.

Requires:
- #952 
- #953 
- #958 
- #961 
- #963 
- #967 

Closes DOV-6868 (eventually).

### CODE REVIEW

File: docs/core/plugins/mail_crypt.md [hunk 1]
```diff
@@ -399,9 +399,9 @@ doveadm \
   -o fs=crypt,posix \
   -o fs/crypt/fs_driver=crypt \
   -o fs/posix/fs_driver=posix \
-  -o crypt_private_key="$(cat pubkey.pem)" \
+  -o crypt_private_key_file=pubkey.pem \
   -o crypt_global_private_key=main \
-  -o crypt_global_private_key/main/crypt_private_key="$(cat privkey.pem)" \
+  -o crypt_global_private_key/main/crypt_private_key_file= privkey.pem \
```
Comment (sirainen) [inline docs/core/plugins/mail_crypt.md:407]: extra space after`=` (which breaks it)

File: docs/core/config/auth/databases/oauth2.md [hunk 2]
```diff
@@ -81,7 +81,10 @@ Without proxy authentication:
 
 ```[dovecot.conf]
 passdb static {
-  args = nopassword=y proxy=y proxy_mech=%m ...
+  nopassword = yes
```
Comment (sirainen) [inline docs/core/config/auth/databases/oauth2.md:84]: These need to be inside fields { .. }

File: docs/core/plugins/mail_crypt.md [hunk 3]
```diff
@@ -322,7 +322,11 @@ crypt_global_private_key main {
 }
 
 passdb static {
-  args = password=pass crypt_global_public_key_file=ecpubkey.pem crypt_global_private_key/main/private_key=<content of ecprivkey.pem>
+  fields {
+    password = pass
```
Comment (sirainen) [inline docs/core/plugins/mail_crypt.md:326]: password outside fields

File: docs/core/plugins/fts_solr.md [hunk 4]
```diff
@@ -154,10 +154,8 @@ mail_plugins {
   fts_solr = yes
 }
 
-plugin {
-  fts = solr
-  fts_solr = url=https://solr.example.org:8983/solr/ debug batch_size=1000
-}
+fts_driver = solr
+fts_solr_url = url=https://solr.example.org:8983/solr/ debug batch_size=1000
```
Comment (sirainen) [inline docs/core/plugins/fts_solr.md]: This isn't right. And I think there aren't any more fts-solr changes after this pull request. These need to be split into separate settings:

fts_solr_url = https://...
fts_solr_batch_size = 1000

fts_solr_debug setting is gone

**Review by sirainen — CHANGES_REQUESTED** [review]

File: docs/core/plugins/fts_solr.md (hunk 4 above)
Comment (fleischie) [reply docs/core/plugins/fts_solr.md]: I changed the commit in Part IV, where this was introduced.

**Pull request merged**